roles/foreman_repositories/tasks/redhat.yml

```diff
@@ -24,3 +24,32 @@
     disable_gpg_check: true
     state: present
   when: foreman_repositories_katello_version is defined
+
+- when:
+    - foreman_repositories_katello_version is defined
+    - ansible_distribution_major_version == '8'
+  block:
+    - name: Install dnf-config-manager
+      package:
+        name: dnf-plugins-core
+        state: present
+
+    - name: Get pki-core module info
+      command: dnf module list --enabled
+      register: enabled_modules
+      changed_when: false
+
+    - name: Enable pki-core module
+      command: dnf module enable pki-core -y
+      when: "'pki-core' not in enabled_modules.stdout"
+
+    - name: List disabled repositories
+      command: dnf repolist --disabled
+      register: disabled_repos
+      changed_when: false
+
+    - name: Enable powertools
+      command: dnf config-manager --set-enabled powertools
```

Comment:
can't this be done with `yum_repository` module or worst case `ini_file`?

Reply:
I dont think it can with `yum_repository` as you have to define the baseurl or mirrorlist yourself. Unless we want to do that rather than just enabling what is there? `ini_file` seems like a weird choice when I can use dnf directly to handle things.

Reply:
ah, indeed, that's a rather stupid limitation of the module

fair enough.